File: dkplayer-java/src/main/java/xyz/doikki/videoplayer/internal/DKPlayerException.java
package xyz.doikki.videoplayer.internal;

import android.annotation.SuppressLint;
import android.media.MediaPlayer;

public class DKPlayerException extends RuntimeException {

    private int what = MediaPlayer.MEDIA_ERROR_UNKNOWN;
    private int extra = MediaPlayer.MEDIA_ERROR_UNKNOWN;

    @SuppressLint("DefaultLocale")
    public DKPlayerException(int what, int extra) {
        super(String.format("what=%d,extra=%d", what, extra));
        this.what = what;
        this.extra = extra;
    }

    public DKPlayerException(Throwable cause) {
        super(cause);
    }

    public int getWhat() {
        return what;
    }

    public int getExtra() {
        return extra;
    }


}

File: dkplayer-java/src/main/java/xyz/doikki/videoplayer/internal/DKPlayerException.kt
package xyz.doikki.videoplayer.internal

import android.media.MediaPlayer
import android.annotation.SuppressLint
import java.lang.RuntimeException

class DKPlayerException : RuntimeException {
    var what = MediaPlayer.MEDIA_ERROR_UNKNOWN
        private set
    var extra = MediaPlayer.MEDIA_ERROR_UNKNOWN
        private set

    @SuppressLint("DefaultLocale")
    constructor(what: Int, extra: Int) : super(String.format("what=%d,extra=%d", what, extra)) {
        this.what = what
        this.extra = extra
    }

    constructor(cause: Throwable?) : super(cause) {}
}